src/client/signatures.rs

```diff
@@ -86,3 +75,184 @@ impl From<CanonicalJSONError> for SignatureError {
         err.into()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{Collection, Verification};
+    use env_logger;
+    use httpmock::Method::GET;
+    use httpmock::{Mock, MockServer};
+    use serde_json::json;
+
+    fn verify_signature(
+        mock_server: &MockServer,
+        collection: Collection,
+        certificate: &str,
+        should_fail: bool,
+    ) {
+        let mut get_pem_certificate = Mock::new()
+            .expect_method(GET)
+            .expect_path(
+                "/chains/remote-settings.content-signature.mozilla.org-2020-09-04-17-16-15.chain",
+            )
+            .return_status(200)
+            .return_header("Content-Type", "application/json")
+            .return_body(certificate)
+            .create_on(&mock_server);
+
+        let mut verifiers: Vec<&dyn Verification> = Vec::new();
+
+        #[cfg(feature = "ring_verifier")]
+        verifiers.push(&super::ring_verifier::RingVerifier {});
+
+        #[cfg(feature = "rc_crypto_verifier")]
+        verifiers.push(&super::rc_crypto_verifier::RcCryptoVerifier {});
+
+        for verifier in &verifiers {
+            if should_fail {
+                assert!(verifier.verify(&collection).is_err())
+            } else {
+                match verifier.verify(&collection) {
+                    Err(err) => {
+                        println!("{:?}", err);
+                        assert!(false)
+                    }
+                    Ok(_) => println!("success"),
+                }
+            }
+        }
+
+        assert_eq!(verifiers.len(), get_pem_certificate.times_called());
+        get_pem_certificate.delete();
+    }
+
+    fn init() {
+        let _ = env_logger::builder().is_test(true).try_init();
+    }
+
+    #[test]
+    fn test_verify_signature() {
+        init();
+
+        let mock_server: MockServer = MockServer::start();
+
+        const VALID_CERTIFICATE: &str = r#"-----BEGIN CERTIFICATE-----
```

Comment:
You can remove a line break from a string literal by escaping it with an back slash. This would allow you to put the `BEGIN CERTIFICATE` on the next line without changing the value of the string literal, in case you prefer that.